File: core/example/stack_example.cpp
#include <iostream>

#include <include/ndpp/ndpp.hpp>

using namespace std;
using namespace ndpp;


int main()
{
//while (1)
//{
    Stack test_stack;

    test_stack.push(6);
    test_stack.push(std::string("abc"));
    test_stack.push(5.0f);
    test_stack.push(ndpp::zeros({3, 4, 5}));

    Stack test_stack2 = test_stack.clone();
    cout << "test_stack's size: " << test_stack.size() << endl;

    for (; !test_stack.empty(); test_stack.pop())
    {
        cout << test_stack.top().name() << endl;
    }

    cout << "============" << endl;

    
    cout << "test_stack2's size: " << test_stack2.size() << endl;
    for (; !test_stack2.empty(); test_stack2.pop())
    {
        cout << test_stack2.top().name() << endl;
    }
//}

    return 0;
}

File: example/python/stack_example.py
import ndpp


if __name__ == "__main__":
    test_stack = ndpp.stack()

    test_stack.push("jj")
    test_stack.push(123)
    test_stack.push({"test": 789})

    copy_tst_stack = test_stack.clone()
    test_stack.pop()

    print("All nodes in test_stack.")
    # show test_stack.
    while test_stack.size():
        print(test_stack.top())
        test_stack.pop()
    
    print("\nAll nodes in copy_tst_stack.")
    # show copy_tst_stack.
    while copy_tst_stack.size():
        print(copy_tst_stack.top())
        copy_tst_stack.pop()